src/rascal/structure_managers/adaptor_increase_maxorder.hh

```diff
@@ -459,28 +430,84 @@ namespace rascal {
 
     internal::for_each(this->cluster_indices_container,
                        internal::ResizePropertyToZero());
 
+    auto & atom_cluster_indices{std::get<0>(this->cluster_indices_container)};
+
     this->nb_neigh.clear();
     this->offsets.clear();
     this->neighbours_atom_tag.clear();
 
-    // #BUG8486@(markus) I now append the ghost atoms to the cluster index
-    // container
     for (auto atom : this->manager) {
-      //  Order 1, but variable Order is at 0, atoms, index 0
-      using AddOrderLoop = AddOrderLoop<atom.order(), traits::NeighbourListType,
-                                        IsCompactCluster,
-                                        atom.order() == (traits::MaxOrder - 1)>;
-
-      auto & atom_cluster_indices{std::get<0>(this->cluster_indices_container)};
-
-      auto indices = atom.get_cluster_indices();
+      // copy the underlying cluster indices of this->manager (not changed)
+      // do order == 1
+      auto indices{atom.get_cluster_indices()};
       atom_cluster_indices.push_back(indices);
-
-      AddOrderLoop::loop(atom, atom, 0, *this);
+      // do the rest
+      AddOrderLoop::loop(atom, *this);
+
+      this->add_entry_number_of_neighbours();
+      std::array<int, traits::MaxOrder> new_tag_list{};
+      // loop over the highest order available in this->manager
+      for (auto cluster :
+           atom.template get_clusters_of_order<traits::MaxOrder - 1>(
+               cluster_start)) {
+        auto && tag_list{cluster.get_atom_tag_list()};
+        // copy the tags from the previous order skiping the center atom tag
+        for (size_t ii{0}; ii < traits::MaxOrder - 1; ++ii) {
+          new_tag_list[ii] = tag_list[ii];
+        }
+        for (auto pair : atom.template get_clusters_of_order<2>()) {
+          // copy to the last element the atom tag of the new order
+          new_tag_list.back() = pair.get_atom_tag();
+          if (traits::NeighbourListType ==
+              AdaptorTraits::NeighbourListType::half) {
+            if (IsCompactCluster) {
+              throw std::runtime_error("Not implemented yet.");
+            } else {
+              // only add strictly lexicographicaly ordered tags, e.g.
+              // (1,2,2) or (1,5,3) are not valid and (1,3,5) is valid
+              if (new_tag_list[traits::MaxOrder - 1] >
+                  new_tag_list[traits::MaxOrder - 2]) {
+                std::array<int, traits::MaxOrder - 1> tags{};
+                for (size_t ii{1}; ii < traits::MaxOrder; ++ii) {
+                  tags[ii - 1] = new_tag_list[ii];
+                }
+                this->add_neighbour_of_cluster(tags);
+              }
+            }
+          } else {
+            if (IsCompactCluster) {
+              throw std::runtime_error("Not implemented yet.");
+            } else {
+              // only add tags_list where none of the tags are equal to one
+              // another, e.g. (3,1,2) is valid but (3,2,2) is not.
```

Comment:
Is (3,2,1) possible?
If we plan to provide a proper _full_ list of triplets, this should be included, right?
Otherwise it is not really a full list?
Or asked differently: What is the full triplet list?
And is the (3,2,2) not entirely avoidable with the offset of the pair list iteration when searching for all the pairs?

Reply:
> Is (3,2,1) possible?

yes it is. The full list of triplets is all possible triplet except for repeated entries like (3,2,2) or (2,2,2).

Reply:
> And is the (3,2,2) not entirely avoidable with the offset of the pair list iteration when searching for all the pairs?

I am not sure. Probably a smarter implementation is possible but I would leave that to another PR if it is actually needed.

Reply:
Can we make an issue out of that to not forget it - or at least be reminded?